### ISSUE TITLE
[JavaScript] Move JSON out of the JavaScript package.

### ISSUE THREAD

**Issue by Thom1729** [sublimehq/Packages#1771]
The JSON syntax definition, tmPreferences, and tests are inside the JavaScript directory. This means that a user can't disable the JavaScript package (e.g. when using a third-party package) without also disabling the core JSON syntax (and vice versa).

JSON had its roots in JavaScript, but it is nowadays used for all sorts of things. I think that it would make sense to promote JSON to its own default package. There is no module named "JSON" in the Package Control channel, so there should be no serious risk of conflict.

Some third-party code could be hardcoding the location of `JSON.sublime-syntax`. If this is a concern, we could leave a stub at that location:

```yaml
%YAML 1.2
---
name: JSON (Compatibility redirect)
scope: ''
hidden: true
contexts:
  main:
    - match: ''
      embed: 'Packages/JSON/JSON.sublime-syntax'
      escape: \B\b # Never
```

**Comment by FichteFoll** [comment]
Also worth thinking about https://github.com/sublimehq/Packages/issues/285 when this is considered.

**Comment by deathaxe** [comment]
All included sublime-syntax files I've seen so far have been used without path. Something like `- include: "blabla.sublime-syntax"`. The only situations I saw paths were in syntax test files.

In case of a stub file, I think `include` would just be enough?

```
%YAML 1.2
---
name: JSON (Compatibility redirect)
scope: ''
hidden: true
contexts:
  main:
    - include: 'Packages/JSON/JSON.sublime-syntax'
```

**Comment by FichteFoll** [comment]
It would still need a base scope, but other than that I suppose it should work.

**Comment by FichteFoll** [comment]
Can be closed since #1805 has been merged.

**Comment by wbond** [comment]
This has been completed
